// pixel-to-svg/lib/d.js
// No dependency

// Public
'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

var _createClass = (function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ('value' in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; })();

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError('Cannot call a class as a function'); } }

var D = (function () {
  function D(x, y) {
    _classCallCheck(this, D);

    this.x = x;
    this.y = y;
    this.width = 1;
    this.height = 1;
  }

  _createClass(D, [{
    key: 'toString',
    value: function toString() {
      var x = this.x;
      var y = this.y;
      var width = this.width;
      var height = this.height;

      // 1pixel = <path d='M0,0 h1 v1 h-1 z'>
      return 'M' + x + ',' + y + 'h' + width + 'v' + height + 'h-' + width + 'Z';
    }
  }]);

  return D;
})();

exports['default'] = D;
module.exports = exports['default'];
